DynamicActiveQuery.php

```diff
@@ -155,12 +201,95 @@ public function createCommand($db = null)
             % (`?) \{
                 ( [a-z_\x7f-\xff][a-z0-9_\x7f-\xff]* (?: \. [^.|\s]+)* )
                 (?:  \| (binary (?:\(\d+\))? | char (?:\(\d+\))? | time (?:\(\d+\))? | datetime (?:\(\d+\))? | date
-                        | decimal (?:\(\d\d?(?:,\d\d?)?\))?  | double (?:\(\d\d?(?:,\d\d?)?\))?
-                        | int(eger)? | (?:un)? signed (?:\int(eger)?)?)  )?
+                        | decimal (?:\(\d\d?(?:,\d\d?)?\))?  | double (?:\(\d\d?,\d\d?\))?
+                        | int(eger)? | (?:un)? signed(?:\s\int(eger)?)?)  )?
             \} \1 %ix
 REGEXP;
         $sql = preg_replace_callback($pattern, $callback, $sql);
 
         return $db->createCommand($sql, $params);
     }
+
+    /**
+     * Wrap all dynamic attributes like {attr.child} to brackets (! !) to prevent escaping
+     * E.g. without this fix attribute {attr.child} will be escaped to `{attr`.`child}`
+     * @return array
+     */
+    private function preProcessDynamicAttributes()
```

Comment:
assuming we need too keep these methods pre, post, wrap, unwrap methods, are you sure they should be private rather than protected? private is fairly unusual